### PR TITLE
refactor!: remove deprecated GeneratedVaadinTextArea

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Removed `GeneratedVaadinTextArea` and updated `TextArea` accordingly.

## Type of change

- Breaking change

### CODE REVIEW

**Comment by DiegoCardoso** [comment]
I don't see that the `GeneratedVaadinTextArea` class has been removed. Is it still needed to keep?

**Comment by web-padawan** [comment]
@DiegoCardoso My bad 🤦‍♂️  Now it is removed.

File: vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextArea.java [hunk 1]
```diff
@@ -63,24 +75,12 @@ public class TextArea extends GeneratedVaadinTextArea<TextArea, String>
...
-     *            initial value is used only if element has no {@code "value"}
-     *            property value, otherwise element {@code "value"} property is
-     *            ignored and the initial value is set
-     */
-    private TextArea(boolean isInitialValueOptional) {
-        super("", "", false, isInitialValueOptional);
+        if (getElement().getProperty("value") == null) {
+            setModelValue("", false);
+            setPresentationValue("");
+        }
```
Comment (DiegoCardoso) [inline vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextArea.java:83]: Previously, this private `TextArea` constructor was pointing to this one on the `GeneratedVaadinTextArea` class:

https://github.com/vaadin/flow-components/blob/8db01ab64eea9b9614505f94c4502cea79e632ca/vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/GeneratedVaadinTextArea.java#L830-L837

which only has a call to `setPresentationValue`.  

Just to clarify if this difference was intentional?

Comment (web-padawan) [reply vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextArea.java]: Thanks, removed. This code shouldn't be needed. See e.g. `RichTextEditor`:

https://github.com/vaadin/flow-components/blob/8db01ab64eea9b9614505f94c4502cea79e632ca/vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java#L116-L118

**Review by DiegoCardoso — APPROVED** [review]

**Pull request merged**